src/Modules/ModulesGraded.jl

```diff
@@ -960,7 +960,9 @@ end
 
 
 function degrees_of_generators(M::SubModuleOfFreeModule{T}; check::Bool=true) where T
-  return map(gen -> degree(gen; check), gens(M))
+  return get_attribute(M, :degrees_of_generators) do
+    map(gen -> degree(gen; check), gens(M))
+  end::Vector{FinGenAbGroupElem} 
```

Comment:
Computing the degrees of the generators over and over again is a waste of resources. I made it an attribute for the moment. We could also turn it into a field of the struct.